packages/flutter/lib/src/painting/continuous_stadium_border.dart

```diff
@@ -0,0 +1,350 @@
+    // This value comes from the website where the other equations and curves
+    // were found
+    // (https://www.paintcodeapp.com/news/code-for-ios-7-rounded-rectangles).
+    const double totalAffectedCornerPixelRatio = 1.52865;
+
+    // The ratio of the radius to the magnitude of pixels on a given side that
+    // are used to construct the two corners.
+    const double minimalUnclippedSideToCornerRadiusRatio = 2.0 * totalAffectedCornerPixelRatio;
+
+    const double minimalEdgeLengthSideToCornerRadiusRatio = 1.0 / minimalUnclippedSideToCornerRadiusRatio;
+
+    // The maximum aspect ratio of the width and height of the given rect before
+    // clamping on one dimension will occur.
+    const double maxEdgeLengthAspectRatio = 1.0 - minimalEdgeLengthSideToCornerRadiusRatio;
+
+    final double rectWidth = rect.width;
+    final double rectHeight = rect.height;
+    final bool widthLessThanHeight = rectWidth < rectHeight;
+    double width = widthLessThanHeight ? rectWidth.clamp(0.0, maxEdgeLengthAspectRatio * rectHeight) : rectWidth;
+    double height = widthLessThanHeight ? rectHeight : rectHeight.clamp(0.0, maxEdgeLengthAspectRatio * rectWidth);
+
+
+    final double centerX = rect.center.dx;
+    final double centerY = rect.center.dy;
+    final double originX = centerX - width / 2.0;
+    final double originY = centerY - height / 2.0;
+    final double minDimension = math.min(width, height);
+    final double radius = minDimension * minimalEdgeLengthSideToCornerRadiusRatio;
+
+    // These equations give the x and y values for each of the 8 mid and corner
+    // points on a rectangle.
+    //
+    // For example, leftX(k) will give the x value on the left side of the shape
+    // that is precisely `k` distance from the left edge of the shape for the
+    // predetermined 'limitedRadius' value.
+    double leftX(double x) { return centerX + x * radius - width / 2; }
+    double rightX(double x) { return centerX - x * radius + width / 2; }
+    double topY(double y) { return centerY + y * radius - height / 2; }
+    double bottomY(double y) { return centerY - y * radius + height / 2; }
+    double bottomMidY(double y) { return originY + height - y * radius; }
+    double leftMidX(double x) { return originX + x * height; }
+    double rightMidX(double x) { return originX + width - x * radius; }
+
+    // An elliptical shape with 2 straight edges and two 180º curves. The width
+    // is greater than the height.
+    //
+    // Code was inspired from the code listed on this website:
+    // https://www.paintcodeapp.com/news/code-for-ios-7-rounded-rectangles
+    //
+    // The shape is drawn from the top midpoint to the upper right hand corner
+    // in a clockwise fashion around to the upper left hand corner.
+    Path bezierStadiumHorizontal () {
+      return Path()
+        ..moveTo(leftX(2.00593972), topY(0.0))
+        ..lineTo(originX + width - 1.52866483 * radius, originY)
+        ..cubicTo(rightX(1.63527834), topY(0.0),
```

Comment:
Same here. If quadratics can be used, I think they're usually more preferred than cubics from Skia's perspective. My anecdotal experience also told me that cubics are more prone to bugs and rendering artifacts than quadratics since Skia has to handle many more edge cases for cubics than quadratics. Skia's major clients such as Chrome and Anroid probably use much more quadratics than cubics which makes quadratics better tested. @reed-at-google : please correct me if my impression were wrong.

Reply:
cubic paths are definitely present in our framework. If they're prone to bugs or rendering artifacts, we should make a concerted effort to remove them from our codebase and remove the API until ready.

Reply:
Oh, cubics definitely passes Skia's quality control which means that it's fast and correct probably 99.99% of the time. It's just that quadratics are even faster and more reliable. So if it's easy to use quadratics instead cubics, we'll prefer quadratics.